tests/Scout/ScoutEngineTest.php

```diff
@@ -0,0 +1,548 @@
+            },
+            array_replace_recursive($defaultPipeline, [
+                [
+                    '$search' => [
+                        'compound' => [
+                            'filter' => [
+                                ['in' => ['path' => 'qux', 'value' => [1, 2]]],
+                                ['in' => ['path' => 'quux', 'value' => [1, 2]]],
+                            ],
+                        ],
+                    ],
+                ],
+            ]),
+        ];
+
+        yield 'where not in conditions without other conditions' => [
+            function () {
+                $builder = new Builder(new SearchableModel(), 'lar');
+                $builder->whereIn('qux', [1, 2]);
+                $builder->whereIn('quux', [1, 2]);
+                $builder->whereNotIn('eaea', [3]);
+
+                return $builder;
+            },
+            array_replace_recursive($defaultPipeline, [
+                [
+                    '$search' => [
+                        'compound' => [
+                            'filter' => [
+                                ['in' => ['path' => 'qux', 'value' => [1, 2]]],
+                                ['in' => ['path' => 'quux', 'value' => [1, 2]]],
+                            ],
+                            'mustNot' => [
+                                ['in' => ['path' => 'eaea', 'value' => [3]]],
+                            ],
+                        ],
+                    ],
+                ],
+            ]),
+        ];
+
+        yield 'empty where in conditions' => [
+            function () {
+                $builder = new Builder(new SearchableModel(), 'lar');
+                $builder->whereIn('qux', [1, 2]);
+                $builder->whereIn('quux', [1, 2]);
+                $builder->whereNotIn('eaea', [3]);
+
+                return $builder;
+            },
+            array_replace_recursive($defaultPipeline, [
+                [
+                    '$search' => [
+                        'compound' => [
+                            'filter' => [
+                                ['in' => ['path' => 'qux', 'value' => [1, 2]]],
+                                ['in' => ['path' => 'quux', 'value' => [1, 2]]],
+                            ],
+                            'mustNot' => [
+                                ['in' => ['path' => 'eaea', 'value' => [3]]],
+                            ],
+                        ],
+                    ],
+                ],
+            ]),
+        ];
+
+        yield 'exclude soft-deleted' => [
+            function () {
+                return new Builder(new SearchableModel(), 'lar', softDelete: true);
+            },
+            array_replace_recursive($defaultPipeline, [
+                [
+                    '$search' => [
+                        'compound' => [
+                            'filter' => [
+                                ['equals' => ['path' => '__soft_deleted', 'value' => 0]],
```

Comment:
Doesn't the value here conflict with the bool cast you do in ScoutEngine? Likewise for the "only trashed" case below.